conans/client/file_copier.py

```diff
@@ -90,10 +90,9 @@ def __call__(self, pattern, dst="", src="", keep_path=True, links=False, symlink
     def _copy(self, base_src, pattern, src, dst, symlinks, ignore_case, excludes, keep_path,
               excluded_folders):
         # Check for ../ patterns and allow them
-        if pattern.startswith(".."):
-            rel_dir = os.path.abspath(os.path.join(base_src, pattern))
-            base_src = os.path.dirname(rel_dir)
-            pattern = os.path.basename(rel_dir)
+        while pattern.startswith(".."):
```

Comment:
this check is not correct as `..` might be a part of the file name (e.g. run `touch ...1`)

Reply:
No it's not, and I believe it wasn't before, or was it? Only if the code was a no-op in such cases, which it isn't in general because it reduces the pattern to just its `basename()`.

It could be made a little more correct by saying
```
        while pattern.startswith(".." + os.sep) or \
            (os.altsep is not None and pattern.startswith(".." + os.altsep)):
```
ugly, but perhaps safer?